### PR TITLE
Add initial VPC configuration API and improve interrupt handling

### PR DESCRIPTION
**Pull request by shadaj**
Add initial VPC configuration API and improve interrupt handling

### CODE REVIEW

**Comment by shadaj** [comment]
@MingweiSamuel would you be able to test out this build on Windows? added some Unix-specific bits so want to make sure I got the cfg flags right

**Comment by MingweiSamuel** [comment]
Sure

**Comment by MingweiSamuel** [comment]
Windows build is good

File: hydro_cli/src/core/gcp.rs [hunk 1]
```diff
@@ -411,7 +446,12 @@ impl Host for GCPComputeEngineHost {
             }),
         );
 
-        let vm_instance = format!("vm-instance-{project}-{id}");
+        let terraform_alphabet: [char; 16] = [
+            '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'a', 'b', 'c', 'd', 'e', 'f',
+        ];
```
Comment (MingweiSamuel) [inline hydro_cli/src/core/gcp.rs:451]: make this a static somewhere instead of duplicating it

File: hydro_cli/src/core/gcp.rs [hunk 2]
```diff
@@ -217,12 +218,46 @@ impl LaunchedHost for LaunchedComputeEngine {
...
+    fn collect_resources(&mut self, resource_batch: &mut ResourceBatch) {
+        if self.vpc_id.is_some() {
+            return;
+        }
+
+        let terraform_alphabet: [char; 16] = [
+            '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'a', 'b', 'c', 'd', 'e', 'f',
+        ];
+
+        let vpc_network = format!("vpc-network-{}", nanoid!(8, &terraform_alphabet));
```
Comment (MingweiSamuel) [inline hydro_cli/src/core/gcp.rs:236]: Make this a static array and/or helper fn

**Review by MingweiSamuel — APPROVED** [review]

**Pull request merged**